app/src/androidTest/java/com/eis/phoneringer/structure/AppManagerTest.java

```diff
@@ -0,0 +1,53 @@
+package com.eis.phoneringer.structure;
+
+import android.content.Context;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import com.eis.phoneringer.exceptions.WrongPasswordException;
+import com.eis.smslibrary.SMSPeer;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class AppManagerTest {
+
+    Context context = null;
+    static AppManager appManager = null;
+    static PasswordManager passwordManager = null;
+    static RingtoneHandler ringtoneHandler = RingtoneHandler.getInstance();
+    static final String WRONG_PASSWORD = "username";
+    static final String VALID_PASSWORD = "password";
+    static final String VALID_PHONE_NUMBER = "+391111111111";
+
+    @Before
+    public void init() {
+        context = InstrumentationRegistry.getInstrumentation().getTargetContext();
+        appManager = AppManager.getInstance();
+        passwordManager = new PasswordManager(context);
+    }
+
+    @Test
+    public void onRingCommandReceived_wrong_password() {
```

Comment:
What about using
`@Test(expected = WrongPasswordException.class)`
and remove the try-catch? The current method of doing assert.fail to assert that exceptions are thrown is good only if exceptions are NOT Runtime, so when you **must** catch them. In this case, where the exception is runtime, we can use the expected.